dataland-frontend/src/utils/UpdateObjectUtils.ts

```diff
@@ -0,0 +1,45 @@
+export type objectType = { [key: string]: string | object };
+/**
+ * Updates keys from one object to another
+ *
+ * @param baseObject object to be updated
+ * @param objectWithNewData object to be updated
+ */
+export function updateObject(baseObject: objectType, objectWithNewData: objectType): void {
+  for (const key in objectWithNewData) {
+    if (typeof objectWithNewData[key] === "object" && objectWithNewData[key] !== null) {
+      if (baseObject[key]) {
+        updateObject(baseObject[key] as unknown as objectType, objectWithNewData[key] as unknown as objectType);
+      } else {
+        baseObject[key] = {};
+        updateObject(baseObject[key] as unknown as objectType, objectWithNewData[key] as unknown as objectType);
+      }
+    } else if (objectWithNewData[key] !== null) {
+      baseObject[key] = objectWithNewData[key];
+    }
+  }
+}
+
+/**
+ * Changes the value of a variable by dividing it by 100
+ *
+ * @param obj object in which it is looking for the value to change
+ * @param typeOfModification determines how we change object values
+ * @returns Object modified
+ */
+export function modifyObjectKeys(obj: objectType, typeOfModification: string): objectType {
+  const objectModified = obj;
+  for (const key in objectModified) {
+    if (key === "value" && objectModified[key]) {
+      if (typeOfModification === "send") {
+        objectModified[key] = (Math.round(+objectModified[key] * 100) / 100 / 100).toString();
+      }
+      if (typeOfModification === "receive") {
+        objectModified[key] = (Math.round(+objectModified[key] * 100 * 100) / 100).toString();
+      }
+    } else if (typeof objectModified[key] === "object" && key !== "totalAmount") {
+      modifyObjectKeys(objectModified[key] as unknown as objectType, typeOfModification);
+    }
+  }
+  return objectModified;
+}
```

Comment:
This is a little confusing as I look at it as an outsider.

1.) The description is incomplete. Only in the case "send" the value is divided by 100. E.g. in the case "receive" the value is multiplied by 100. And in other cases it will dig deeper into the object recursively.

2.) Isn't this method a "conversion-layer" to convert frontend values to backend model values and vice versa?
In that case you can also explain this in the description and also change the name of the method to something more specific, e.g. `convertNumberValuesInObject`

3,) If `typeOfModification` has to contain specific values "send" and "receiver" for something to actually happen, then I would suggest making it of type `"send" | "receive"` so someone who reads it for the first time gets it right away.

Reply:
Yes, right. Changed